typescript/secrets-manager-secret-rotation/index.ts

```diff
@@ -87,70 +112,79 @@ export class SecretsManagerCustomRotationStack extends Stack {
-        resources: ["arn:aws:elasticache:"+Stack.of(this).region+":"+Stack.of(this).account+":replicationgroup:"+ecClusterReplicationGroup.replicationGroupId],
+        resources: [
+          "arn:aws:elasticache:" +
+            Stack.of(this).region +
+            ":" +
+            Stack.of(this).account +
+            ":replicationgroup:" +
+            ecClusterReplicationGroup.replicationGroupId,
+        ],
         actions: [
           "elasticache:ModifyReplicationGroup",
-          "elasticache:DescribeReplicationGroups"
-        ]
+          "elasticache:DescribeReplicationGroups",
+        ],
       })
     );
 
     rotatorRole.addToPolicy(
       new iam.PolicyStatement({
         effect: iam.Effect.ALLOW,
         resources: ["*"],
-        actions: [
-          "secretsmanager:GetRandomPassword"
-        ]
+        actions: ["secretsmanager:GetRandomPassword"],
       })
     );
 
-    const redisPyLayer = new lambda.LayerVersion(this, 'redispy_Layer', {
-      code: lambda.Code.fromAsset(path.join(__dirname, 'lambda/lib/redis_module/redis_py.zip')),
-      compatibleRuntimes: [lambda.Runtime.PYTHON_3_8, lambda.Runtime.PYTHON_3_7, lambda.Runtime.PYTHON_3_6],
-      description: 'A layer that contains the redispy module',
-      license: 'MIT License'
+    const redisPyLayer = new pyLambda.PythonLayerVersion(this, "RedisPyLayer", {
+      entry: path.join(__dirname, "lambda", "layer", "redis-py"),
+      compatibleRuntimes: [
+        lambda.Runtime.PYTHON_3_8,
+        lambda.Runtime.PYTHON_3_7,
+        lambda.Runtime.PYTHON_3_6,
+      ],
+      description: "A layer that contains the redis-py module",
+      license: "MIT License",
     });
 
-    const fn = new lambda.Function(this, 'function', {
+    const fn = new pyLambda.PythonFunction(this, "SecretRotationFunction", {
       runtime: lambda.Runtime.PYTHON_3_7,
-      handler: 'lambda_handler.lambda_handler',
-      code: lambda.Code.fromAsset(path.join(__dirname, 'lambda')),
+      entry: path.join(__dirname, "lambda"),
+      handler: "lambda_handler.lambda_handler",
       layers: [redisPyLayer],
       role: rotatorRole,
       timeout: Duration.seconds(30),
       vpc: vpc,
-      vpcSubnets: {subnetType: ec2.SubnetType.PRIVATE_WITH_NAT},
+      vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_NAT },
       securityGroups: [ecSecurityGroup, rotatorSecurityGroup],
       environment: {
         replicationGroupId: ecClusterReplicationGroup.ref,
         redis_endpoint: ecClusterReplicationGroup.attrPrimaryEndPointAddress,
         redis_port: ecClusterReplicationGroup.attrPrimaryEndPointPort,
-        EXCLUDE_CHARACTERS: '@%*()_+=`~{}|[]\\:";\'?,./',
-        SECRETS_MANAGER_ENDPOINT: "https://secretsmanager."+Stack.of(this).region+".amazonaws.com"
-      }
+        EXCLUDE_CHARACTERS: "@%*()_+=`~{}|[]\\:\";'?,./",
+        SECRETS_MANAGER_ENDPOINT:
+          "https://secretsmanager." + Stack.of(this).region + ".amazonaws.com",
+      },
```

Comment:
Apologies for the messy diff introduced by my formatter. L146-L174 and the newly imported `pyLambda` module are the only functional changes.